src/osd/ECBackend.cc

```diff
@@ -500,6 +500,8 @@ void ECBackend::continue_recovery_op(
 	assert(!op.recovery_progress.first);
 	dout(10) << __func__ << ": canceling recovery op for obj " << op.hoid
 		 << dendl;
+	// XXX: Doing this on the read error handling just loops in recovery
+	// How would this case ever actually happen?
```

Comment:
You added this code in d9106ce5e4437ab02279c4d7dfa23ce0d69c59df to fix tracker 8161.  I tried to use an equivalent version in OnRecoveryReadComplete::finish() only to discovery that it didn't work.  I'll have to look into whether I can do something like my new _failed_push() to handle this case too.